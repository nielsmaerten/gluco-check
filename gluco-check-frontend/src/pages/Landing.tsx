import React from "react";
import { useTranslation } from "react-i18next";
import { Container, Grid, makeStyles, Typography } from "@material-ui/core";
import Onboarding from "../components/Onboarding";

import * as firebase from "firebase/app";
import * as firebaseui from "firebaseui";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import { auth } from "../lib/firebase";
import Boilerplate from "../components/Boilerplate";

const useStyles = makeStyles((theme) => ({
  container: {
    paddingTop: theme.spacing(2),
    textAlign: "center",
  },
  subtitle: {
    marginBottom: theme.spacing(3),
    textAlign: "center",
    [theme.breakpoints.down("sm")]: {
      marginBottom: 0,
      "& h2": {
        fontSize: "1.125rem",
        lineHeight: 1.6,
      },
    },
  },
  contentGrid: {
    minHeight: "80vh",
  },
  utterance: {
    fontWeight: "bold",
  },
  boilerplate: {
    textAlign: "center",
    marginTop: theme.spacing(3),
    "& p": {
      marginBottom: theme.spacing(1),
      [theme.breakpoints.down("sm")]: {
        fontSize: ".7rem",
      },
    },
  },
  ctaContainer: {
    "&.MuiGrid-item": {
      padding: "0",
    },
  },
}));

function Landing() {
  const classes = useStyles();
  const { t } = useTranslation();

  const firebaseUIConfig = {
    credentialHelper: firebaseui.auth.CredentialHelper.NONE,
    signInFlow: "redirect",
    signInSuccessUrl: "/",
    signInOptions: [
      {
        provider: firebase.auth.GoogleAuthProvider.PROVIDER_ID,
        fullLabel: t("login.buttonLabel"),
      },
    ],
  };

  return (
    <>
      <Grid
        className={classes.contentGrid}
        container
        direction="column"
        justify="center"
        alignItems="center"
        spacing={2}
        wrap="nowrap"
      >
        <Grid item className={classes.subtitle}>
          <Container maxWidth="sm">
            <Typography variant="h5" component="h2">
              {t("landing.subtitle")}
            </Typography>
          </Container>
        </Grid>
        <Grid item>
          <Onboarding />
        </Grid>
        <Grid item className={classes.ctaContainer}>
          <StyledFirebaseAuth uiConfig={firebaseUIConfig} firebaseAuth={auth} />
        </Grid>
        <Grid item className={classes.boilerplate}>
          <Boilerplate />
        </Grid>
      </Grid>
    </>
  );
}

export default Landing;
