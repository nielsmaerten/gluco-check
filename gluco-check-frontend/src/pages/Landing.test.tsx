import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import Landing from "./Landing";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
}));

const mockLanguage = "en";
jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
      i18n: {
        language: mockLanguage,
      },
    };
  },
}));

jest.mock("../lib/firebase");
jest.mock("react-firebaseui/StyledFirebaseAuth", () => {
  return {
    __esModule: true,
    default: () => {
      return <>This is where the login would be</>;
    },
  };
});

jest.mock("../components/Onboarding", () => {
  return {
    __esModule: true,
    default: () => {
      return <>Onboarding</>;
    },
  };
});

jest.mock("../components/Boilerplate", () => {
  return {
    __esModule: true,
    default: () => {
      return <>Boilerplate</>;
    },
  };
});

afterEach(cleanup);

describe("Landing page", () => {
  it("renders the component", () => {
    const { container } = render(<Landing />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Landing />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
