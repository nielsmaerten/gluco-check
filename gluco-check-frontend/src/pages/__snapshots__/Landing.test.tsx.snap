// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page renders the component 1`] = `
<div
  class="MuiGrid-root makeStyles-contentGrid-3 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root makeStyles-subtitle-2 MuiGrid-item"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        landing.subtitle
      </h2>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    Onboarding
  </div>
  <div
    class="MuiGrid-root makeStyles-ctaContainer-6 MuiGrid-item"
  >
    This is where the login would be
  </div>
  <div
    class="MuiGrid-root makeStyles-boilerplate-5 MuiGrid-item"
  >
    Boilerplate
  </div>
</div>
`;
